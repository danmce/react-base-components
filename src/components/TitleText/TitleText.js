import React from 'react'
import PropTypes from 'prop-types'
import StyledTitleText from './TitleText.styles'

const TitleText = ({ children }) => (
  <StyledTitleText>{children}</StyledTitleText>
)

TitleText.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
}

export default TitleText
