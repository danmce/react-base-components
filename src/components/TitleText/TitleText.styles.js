import styled from '@emotion/styled'

export default styled.h1`
  font-family: ${props => props.theme.typography.fontFamily};
  font-size: ${props => props.theme.typography.size.xl}px;
`
