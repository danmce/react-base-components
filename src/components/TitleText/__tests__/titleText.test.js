import React from 'react'
import * as enzyme from 'enzyme'
import toJson from 'enzyme-to-json'
import TitleText from '../TitleText'
import UxPinWrapper from '../../../wrappers/UXPinWrapper'

test('renders Typography correctly', () => {
  const bodyText = (
    <UxPinWrapper>
      <TitleText>Title Text</TitleText>
    </UxPinWrapper>
  )

  expect(toJson(enzyme.mount(bodyText))).toMatchSnapshot('enzyme.mount')
})
