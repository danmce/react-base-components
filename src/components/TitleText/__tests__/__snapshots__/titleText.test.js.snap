// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Typography correctly: enzyme.mount 1`] = `
.emotion-0 {
  font-family: 'Helvetica Neue',sans-serif;
  font-size: 24px;
}

<UXPinWrapper>
  <ThemeProvider
    theme={
      Object {
        "borders": Object {
          "borderRadius": "3px",
          "borderStandard": "1px solid",
          "default": Object {
            "borderRadius": "3px",
            "borderStandard": "1px solid",
          },
        },
        "colors": Object {
          "button": Object {
            "error": Object {
              "case": "none",
              "fill": undefined,
              "stroke": undefined,
              "tint": "#FFFFFF",
            },
            "primary": Object {
              "case": "none",
              "fill": "#17384B",
              "stroke": "#2C546C",
              "tint": "#FFFFFF",
            },
            "secondary": Object {
              "case": "none",
              "fill": "transparent",
              "stroke": "#2C546C",
              "tint": "#17384B",
            },
            "success": Object {
              "case": "none",
              "fill": undefined,
              "stroke": undefined,
              "tint": "#FFFFFF",
            },
            "warning": Object {
              "case": "none",
              "fill": undefined,
              "stroke": undefined,
              "tint": "#FFFFFF",
            },
          },
        },
        "typography": Object {
          "fontFamily": "'Helvetica Neue', sans-serif",
          "lineHeightMultiplier": Object {},
          "size": Object {
            "l": 16,
            "m": 12,
            "s": 10,
            "xl": 24,
            "xs": 8,
            "xxl": 34,
            "xxxl": 50,
          },
          "weight": Object {
            "bold": 600,
            "light": 100,
            "regular": 300,
          },
        },
      }
    }
  >
    <TitleText>
      <Styled(h1)>
        <h1
          className="emotion-0 emotion-1"
        >
          Title Text
        </h1>
      </Styled(h1)>
    </TitleText>
  </ThemeProvider>
</UXPinWrapper>
`;
